Libraries/dotNetRDF/Storage/StardogConnector.cs

```diff
@@ -1932,10 +1932,10 @@ public virtual void Query(string query, AsyncStorageCallback callback, object st
                 }
             }
 
-            requestUri += GetReasoningParameter();
+            if (Reasoning != StardogReasoningMode.DatabaseControlled) requestUri += GetReasoningParameter();
```

Comment:
This isn't really needed as GetReasoningParameter returns an empty string if ReasoningMode is set to DatabaseControlled. I'd rather leave the logic of setting the request parameter encapsulated in that method than add an exception here.

Reply:
Reverted back to the original logic